File: Week 1/reverse_array.py
def reverse(a,start,end):
    while start < end :
        a[start],a[end] = a[end],a[start]
        start +=1
        end -=1


a = [1,2,3,4,5,6]
print("Array= ",a)
reverse(a,0,5)
print('Reversed Array= ', a)

File: Week 1/reverse_array.cpp
#include<iostream>
using namespace std;

void reverse(int arr1[], int start, int end)
{
    while (start < end)
    {
        int temp = arr1[start]; 
        arr1[start] = arr1[end];
        arr1[end] = temp;
        start++;
        end--;
    } 
}

int main()
{
	int n,arr[n];
	cout<<"Enter Length of an Array :";
	cin>>n;
	for(int i=0;i<n;i++)
	{
		cin>>arr[i];
	}
	
	reverse(arr,0,n-1);
	
	for(int i=0;i<n;i++)
	{
		cout<<arr[i]<<" ";
	}
	return 0;
}
